### PR TITLE
Add skips and comments for certain mutants

### PR DESCRIPTION
**Pull request by ASuciuX**
- migrated this https://github.com/stacks-network/stacks-core/pull/4604 as it was on top of next to this one

Related issue #4587 

List of PRs addressed:
- [x] #4412
- [x] #4414 
- [x] #4527
- [x] #4530
- [x] #4556 
- [x] #4560 
- [x] #4561 
- [x] #4575 
- [x] #4576 
- [x] #4578

### CODE REVIEW

**Review by jbencin — DISMISSED** [review]

File: stackslib/Cargo.toml [hunk 1]
```diff
@@ -93,6 +93,7 @@ features = ["serde"]
 [dependencies.time]
 version = "0.2.23"
 features = ["std"]
+mutants = "0.0.3"
```
Comment (jcnelson) [inline stackslib/Cargo.toml:96]: This should either be a dev-dependency, or an optional and off-by-default dependency

File: clarity/src/vm/ast/parser/v2/mod.rs [hunk 2]
```diff
@@ -206,6 +206,16 @@ impl<'a> Parser<'a> {
         }
     }
 
+    // TODO: #4587 create default for `SymbolicExpression`, then check if mutation tests are caught for these cases:
```
Comment (jcnelson) [inline clarity/src/vm/ast/parser/v2/mod.rs:209]: That applies here as well, and in all the other places in this PR (I'm not going to tag them all).

File: libsigner/Cargo.toml [hunk 3]
```diff
@@ -33,6 +33,7 @@ stackslib = { path = "../stackslib"}
 thiserror = "1.0"
 tiny_http = "0.12"
 wsts = { workspace = true }
+mutants = "0.0.3"
```
Comment (jcnelson) [inline libsigner/Cargo.toml:36]: Why is this now a dependency?  We shouldn't be linking it into the production binary.

File: pox-locking/Cargo.toml [hunk 4]
```diff
@@ -22,6 +22,7 @@ path = "src/lib.rs"
 clarity = { package = "clarity", path = "../clarity" }
 stacks_common = { package = "stacks-common", path = "../stacks-common" }
 slog = { version = "2.5.2", features = [ "max_level_trace" ] }
+mutants = "0.0.3"
```
Comment (jcnelson) [inline pox-locking/Cargo.toml:25]: Same here -- `mutants` is a dev-dependency, or its an optional and off-by-default dependency in all crates.

**Review by jcnelson — COMMENTED** [review]
Thanks for this @ASuciuX!  Please address the comments and I'll approve.

File: clarity/src/vm/ast/parser/v1.rs [hunk 5]
```diff
@@ -508,6 +508,16 @@ fn handle_expression(
     }
 }
 
+// TODO: #4587 create default for `SymbolicExpression`, then check if mutation tests are caught for these cases:
```
[the remark this answers is not in the document]
Comment (ASuciuX) [reply clarity/src/vm/ast/parser/v1.rs:511]: Got it, thanks @jcnelson . Best structure would be 1 issue for each `TODO comment` function and have all needed tests into that 1 issue? I will create a label to be related to mutant testing as there will some quite a few issues created so it's easier to find them afterwards. Not sure how should I link them to the code here? Should I paste the issue link here? If I would link the code inside the issue there might be the chance it gets moved before it gets fixed.
```
// ParseResult::from_iter([Some(Default::default())])
// ParseResult::new(None)
// ParseResult::from_iter([None])
// ParseResult::new(Some(Default::default()))
// ParseResult::from(None)
// ParseResult::from(Some(Default::default()))
// ParseResult::new()
// Or keep the skip and remove the comment
 ```

File: clarity/src/vm/ast/parser/v1.rs [hunk 6]
```diff
@@ -508,15 +508,7 @@ fn handle_expression(
     }
 }
 
-// TODO: #4587 create default for `SymbolicExpression`, then check if mutation tests are caught for these cases:
```
Comment (wileyj) [inline clarity/src/vm/ast/parser/v1.rs]: should this be removed since #4587 hasn't been merged yet?

File: stackslib/Cargo.toml (hunk 1 above)
Comment (ASuciuX) [reply stackslib/Cargo.toml]: moved it to dev-dependency

File: libsigner/Cargo.toml (hunk 3 above)
Comment (ASuciuX) [reply libsigner/Cargo.toml]: moved it to dev-dependency

File: pox-locking/Cargo.toml (hunk 4 above)
Comment (ASuciuX) [reply pox-locking/Cargo.toml:27]: moved it to dev-dependency and on all the other places

File: clarity/src/vm/ast/parser/v1.rs (hunk 5 above)
[the remark this answers is not in the document]
Comment (ASuciuX) [reply clarity/src/vm/ast/parser/v1.rs]: Migrated them to these https://github.com/stacks-network/stacks-core/labels/mutation-testing

**Review by jcnelson — APPROVED** [review]

**Review by jbencin — APPROVED** [review]

**Pull request merged**